data_processing/wdl/tasks/methylationCheck.wdl

```diff
@@ -0,0 +1,49 @@
+version 1.0
+
+task runMethylationCheck {
+    input {
+        Array[File] bam_files
+        String sample_id
+    }
+
+
+     command <<<
+        set -e
+        echo "Starting the task" || exit 1
+        for bam_file in "~{sep=' ' bam_files}"; do
+            echo "Processing $bam_file" || exit 1
+            bam_basename=$(basename $bam_file .bam)
+            methylation_output_file="${bam_basename}_methylation_summary.txt"
+            if samtools view $bam_file 2>/dev/null | grep -m 1 "MM:Z:C+m" > /dev/null; then
+                echo "$bam_file: yes" > $methylation_output_file
+            else
+                echo "$bam_file: no" > $methylation_output_file
+            fi || exit 1
+        done
+    >>>
+
+    runtime {
+        docker: "quay.io/biocontainers/samtools:1.20--h50ea8bc_0"
+    }
+
+    output {
+        Array[File] methylation_outputs = glob("*_methylation_summary.txt")
+    }
+}
+
+workflow methylationWorkflow {
+    input {
+        Array[File] bam_files
+        String sample_id
+    }
```

Comment:
by convention we put the workflow at the top of the WDL. That way people who are reading can orient themselves easily. Doesn't matter too much here, but it's easier to keep the convention in tact.